proposals/0054-sysvar-get-active-stake.md

```diff
@@ -0,0 +1,64 @@
+---
+simd: '0054'
+title: Sysvar for active stake
+authors:
+  - x19
+category: Standard
+type: Core
+status: Draft
+created: 2023-06-7
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+We propose to add a new sysvar that contains vote account pubkeys and 
+their corresponding active stake.
+
+## Motivation
+
+Currently, if a validator wants to prove its active stake to a program, 
+it needs to 
+pass in all of the stake accounts which have delegated to it. This is 
+infeasible due to the large number of stake accounts this would require.
+
+Using the proposed sysvar, a program can look up the corresponding 
+vote account and verify the amount of active stake it has delegated to it.
+
+This sysvar would unlock new use cases which use stake amount in their logic 
+including on-chain governance, attestations, and more.
+
+## Alternatives Considered
+
+None
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+- sysvar structure: `Vec<(vote_account: Pubkey, active_stake_in_lamports: u64)>`
+- sysvar address: `SysvarStakeWeight11111111111111111111111111`
+
```

Comment:
There's a couple additional considerations that I see:

0. The sort order of the sysvar should be defined here: accending by `vote_account` probably. Also lets document that there will be never be any duplicate `vote_account` entries
1. Serialization format of the data. bincode seems like it should be ok: a `u64` for the Vec length, followed by 40 bytes per entry (32 bytes for Pubkey, 8 for active stake)
2. The maximum data size of the sysvar account. Unfortunately I think we do need a limit, perhaps 4096 entries accounts? With bincode format, max data size would be (8 + 40 * 4096) = 163,848 bytes
3. The new sysvar will need 1.14127296 SOL to maintain the rent-exempt minimum balance (`$ solana rent 163848`). Not sure yet where this SOL should come from but I don't think the protocol should just create it when the feature is activated.

Reply:
3. Since we do have  a maximum data size, we'll need to define what happens if there are more than that number of vote accounts (lop off the ones with the least stake)